Explorer/Assets/DCL/SceneLoadingScreens/SceneLoadingScreenController.cs

```diff
@@ -0,0 +1,164 @@
+                ShowTip(currentTip + 1);
+                tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+            });
+
+            viewInstance.ShowPreviousButton.onClick.AddListener(() =>
+            {
+                ShowTip(currentTip - 1);
+                tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+            });
+
+            progressLabel = (IntVariable)viewInstance.ProgressLabel.StringReference["progressValue"];
+        }
+
+        protected override void OnBeforeViewShow()
+        {
+            base.OnBeforeViewShow();
+
+            viewInstance.ProgressBar.normalizedValue = 0f;
+            progressLabel!.Value = 0;
+            viewInstance.ClearTips();
+        }
+
+        protected override void OnViewClose()
+        {
+            base.OnViewClose();
+
+            tipsRotationCancellationToken?.SafeCancelAndDispose();
+        }
+
+        protected override UniTask WaitForCloseIntent(CancellationToken ct)
+        {
+            return ShowTipsAsync(ct)
+                  .ContinueWith(() =>
+                   {
+                       tipsRotationCancellationToken = tipsRotationCancellationToken.SafeRestart();
+                       RotateTipsOverTime(tips.Duration, tipsRotationCancellationToken.Token).Forget();
+                   })
+                  .ContinueWith(() => WaitUntilWorldIsLoadedAsync(ct));
+        }
+
+        private async UniTask ShowTipsAsync(CancellationToken ct)
+        {
+            tips = await sceneTipsProvider.Get(ct);
+
+            List<SceneTips.Tip> list = ListPool<SceneTips.Tip>.Get();
+
+            if (tips.Random)
+                tips.Tips.Shuffle(list);
+            else
+                list.AddRange(tips.Tips);
+
+            foreach (SceneTips.Tip tip in list)
+                viewInstance.AddTip(tip);
+
+            currentTip = 0;
+            viewInstance.ShowTip(currentTip);
+
+            ListPool<SceneTips.Tip>.Release(list);
+        }
+
+        private async UniTask WaitUntilWorldIsLoadedAsync(CancellationToken ct)
+        {
+            async UniTaskVoid UpdateProgressBarAsync(CancellationToken ct)
+            {
+                do
+                {
+                    try
+                    {
+                        float progress = Mathf.Clamp01(await inputData.AsyncLoadProcessReport.ProgressCounter.WaitAsync(ct));
+                        await UniTask.SwitchToMainThread(ct);
+                        viewInstance.ProgressBar.normalizedValue = progress;
+                        progressLabel!.Value = (int)(progress * 100);
+                    }
+                    catch (OperationCanceledException) { }
+                }
+                while (viewInstance.ProgressBar.normalizedValue < 1);
```

Comment:
You don't need to check the progress manually as you provide a cancellation token directly coupled with the length of the process. You can simplify the code like this:
```
                try
                {
                    await foreach (float progress in inputData.AsyncLoadProcessReport.ProgressCounter.WithCancellation(ct))
                    {
                        await UniTask.SwitchToMainThread(ct);
                        viewInstance.ProgressBar.normalizedValue = progress;
                        progressLabel!.Value = (int)(progress * 100);
                    }
                }
                catch (OperationCanceledException) { }
```